metis-indexing/src/main/java/eu/europeana/indexing/tiers/RecordTierCalculationViewGenerator.java

```diff
@@ -0,0 +1,91 @@
+package eu.europeana.indexing.tiers;
+
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.indexing.exception.IndexingException;
+import eu.europeana.indexing.exception.TierCalculationException;
+import eu.europeana.indexing.fullbean.RdfToFullBeanConverter;
+import eu.europeana.indexing.solr.EdmLabel;
+import eu.europeana.indexing.tiers.view.RecordTierCalculationSummary;
+import eu.europeana.indexing.tiers.view.RecordTierCalculationView;
+import eu.europeana.indexing.utils.RdfTierUtils;
+import eu.europeana.indexing.utils.RdfWrapper;
+import eu.europeana.indexing.utils.SolrTier;
+import eu.europeana.metis.schema.convert.RdfConversionUtils;
+import eu.europeana.metis.schema.convert.SerializationException;
+import eu.europeana.metis.schema.jibx.RDF;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RecordTierCalculationViewGenerator {
+
+  private final String europeanaId;
+  private final String providerId;
+  private final String stringRdf;
+
+  public RecordTierCalculationViewGenerator(String europeanaId, String providerId, String stringRdf) {
+    this.europeanaId = europeanaId;
+    this.providerId = providerId;
+    this.stringRdf = stringRdf;
+  }
+
+  public RecordTierCalculationView generate() {
+    //Create an object that has predefined "fake" values
+    RecordTierCalculationView recordTierCalculationView = FakeTierCalculationProvider.getFakeObject();
+    final RecordTierCalculationSummary recordTierCalculationSummary = recordTierCalculationView.getRecordTierCalculationSummary();
+    final TierValues tierValues = calculateTierValues(stringRdf);
+
+    //Update only the summary which is what we need for now
+    // TODO: 04/01/2022 Update the remaining values
+    recordTierCalculationSummary.setEuropeanaRecordId(europeanaId); // Here use XmlRecordProcessorServiceImpl
+    recordTierCalculationSummary.setProviderRecordId(providerId);
+    recordTierCalculationSummary.setContentTier(tierValues.getContentTier());
+    recordTierCalculationSummary.setMetadataTier(tierValues.getMetadataTier());
+    recordTierCalculationSummary.setPortalLink("https://example.com"); // We need the root link from the configuration
+    recordTierCalculationSummary.setHarvestedRecordLink("https://example.com"); // We need a proxy controller for this
+
+    return recordTierCalculationView;
+  }
+
+  private TierValues calculateTierValues(String xml) {
```

Comment:
same here. ```final``` parameters.

Reply:
Check https://github.com/europeana/metis-framework/pull/493#discussion_r781886073
Here it makes a bit more sense to add it which I will do, but still not hugely impact-full since it's quite clear what's going on I think(obviously always depends on interpretation :smile: ).

Reply:
in this case if it helps to keep inmutability. is fine.